src/libxtp/jobcalculators/qmmm.cc

```diff
@@ -45,8 +46,21 @@ void QMMM::ParseSpecificOptions(const tools::Property& options) {
   max_iterations_ = options.get(".max_iterations").as<Index>();
   regions_def_.second = options.get(".regions");
   regions_def_.first = mapfile_;
+  use_gs_for_ex_ = options.get(".use_gs_for_ex").as<bool>();
 
   states_ = options.get(".io_states").as<std::vector<QMState>>();
+  whichSegments_ = options.get(".segments").as<std::string>();
+
+  if (whichSegments_ == "all") {
+    all_segments_ = true;
+  } else {
+    all_segments_ = false;
+    std::stringstream ss(whichSegments_);
+    Index segID;
+    while (ss >> segID) {
+      segments_.push_back(segID);
```

Comment:
Can you use the indexparser class for this it allows you also to specify ranges of indeces e.g. `1 3:9 13` and will convert that to a nice list of indeces.